### PR TITLE
Add api endpoint to create new competencies

### LINKED ISSUE #274: An API endpoint is needed for creating new competencies
## Expected behaviour

Sending an HTTP `POST` request to `/competencies` stores the data from the request body in the database.

## Actual behaviour

There is no API endpoint for creating competencies.

## Details and resources

When a competency is created by a user, it is automatically linked to that user's personal competency model.

Incoming data must be validated.

## Checklist

- [x] This issue is linked to the appropriate project.
- [x] This issue has appropriate labels.

### PR DESCRIPTION
**Pull request by pthan1**
## Proposed changes
- Add and test a function to insert a given competency into the `competencies` table and to the authoring principal's `model_competencies` table.
- Add and test POST method to the `/competencies` API endpoint.

Resolves #274   

## Checklist

- [X] Are the issues being addressed linked to this PR?
- [X] Do all commit messages start with the issue number?
- [X] Are all code changes sufficiently tested?
- [X] Are there screenshots for UI changes?

### CODE REVIEW

**Review by BRashad — APPROVED** [review]
It looks good to me.

File: api/src/services/__tests__/competenciesService.ts [hunk 1]
```diff
@@ -39,16 +39,24 @@ describe('competenciesService', () => {
...
         'insert into `competencies` (`description`, `label`, `principal_id`) values (?, ?, ?)',
         [description, label, principalId],
         [competencyId]
       );
+      mockQuery(
+        'insert into `model_competencies` (`competency_id`, `principal_id`) values (?, ?)',
+        [competencyId, principalId],
+        [model_competency_id]
+      );
+      mockQuery('COMMIT;');
```
Comment (BRashad) [inline api/src/services/__tests__/competenciesService.ts:59]: i think there is an extra semi-colon on line 59 'COMMIT;' ?

**Review by BRashad — APPROVED** [review]

**Review by BRashad — APPROVED** [review]

**Review by BRashad — APPROVED** [review]

File: api/src/services/__tests__/competenciesService.ts [hunk 2]
```diff
@@ -39,16 +39,24 @@ describe('competenciesService', () => {
...
-    it('should create a given competency in the system', async () => {
+    it('should create a competency and model competency in a transaction', async () => {
       const principalId = 2;
       const label = 'label';
       const description = 'description';
       const competencyId = 3;
+      const model_competency_id = 4;
+      mockQuery('BEGIN;');
       mockQuery(
         'insert into `competencies` (`description`, `label`, `principal_id`) values (?, ?, ?)',
```
Comment (eileenxue) [inline api/src/services/__tests__/competenciesService.ts:50]: I suppose this doesn't break the code as long as the query values are referenced in the same order, but I'm wondering if there should be some consistencies of how the column names are listed out? In some areas it's `principalId`, then `label` and `description`, and then in some areas, it differs. What do you think?

**Review by eileenxue — APPROVED** [review]
After reviewing the code and comparing it with the existing examples in the project, I believe it's good to go!

File: api/src/services/__tests__/competenciesService.ts (hunk 2 above)
Comment (pthan1) [reply api/src/services/__tests__/competenciesService.ts:50]: @eileenxue Thanks for pointing that out!  I originally wrote the query in the order that you mentioned.  However, when I ran the tests, I kept on getting "Unexpected Query" error messages.  I then figured out that the columns should be listed alphabetically.  Other `insert` queries in the codebase follow that order too

File: api/src/services/__tests__/competenciesService.ts (hunk 1 above)
Comment (pthan1) [reply api/src/services/__tests__/competenciesService.ts:59]: The semi-colon delineates the end of a query.  I follow the convention of previous mock transactions in the code.

Comment (BRashad) [reply api/src/services/__tests__/competenciesService.ts:59]: ok good

File: api/src/middleware/competenciesRouter.ts [hunk 3]
```diff
@@ -25,4 +27,25 @@ competenciesRouter.get('/', parsePaginationParams(), async (req, res, next) => {
   res.json(collectionEnvelope(competencies, competenciesCount));
 });
 
+competenciesRouter.post('/', async (req, res, next) => {
+  const { principalId } = req.session;
+  const { label: label, description: description } = req.body;
```
Comment (davidvandusen) [inline api/src/middleware/competenciesRouter.ts:32]: Shorthands can be used here. There's no need to alias a property to a new name when the new name is the same as the old name.

File: api/src/services/__tests__/competenciesService.ts (hunk 2 above)
Comment (eileenxue) [reply api/src/services/__tests__/competenciesService.ts:50]: Oh, that's good to know. Thanks for that insightful response, I learned something new!

**Review by davidvandusen — APPROVED** [review]
Only one nitpick.

File: api/src/middleware/competenciesRouter.ts (hunk 3 above)
Comment (pthan1) [reply api/src/middleware/competenciesRouter.ts:32]: Good catch!

**Pull request merged**